src/pugixml.cpp

```diff
@@ -5329,11 +5329,16 @@ namespace pugi
 		return impl::strcpy_insitu(_attr->name, _attr->header, impl::xml_memory_page_name_allocated_mask, rhs, impl::strlength(rhs));
 	}
 
-	PUGI__FN bool xml_attribute::set_value(const char_t* rhs)
+	PUGI__FN bool xml_attribute::set_value(const char_t* rhs, size_t sz)
 	{
 		if (!_attr) return false;
 
-		return impl::strcpy_insitu(_attr->value, _attr->header, impl::xml_memory_page_value_allocated_mask, rhs, impl::strlength(rhs));
+		return impl::strcpy_insitu(_attr->value, _attr->header, impl::xml_memory_page_value_allocated_mask, rhs, sz);
+	}
+
+	PUGI__FN bool xml_attribute::set_value(const char_t* rhs)
+	{
+		return set_value(rhs, impl::strlength(rhs));
```

Comment:
Should this have the original code as is? Alternatively, maybe other set_value methods should be calling the new overload with size? The asymmetry is odd.

Reply:
You are absolutely right.This was never intended... Now I have changed it.

Reply:
Are you sure this was addressed?